presto-main/src/main/java/com/facebook/presto/operator/TableFinishOperator.java

```diff
@@ -136,16 +177,62 @@ public void addInput(Page page)
                 fragmentBuilder.add(VARBINARY.getSlice(fragmentBlock, position));
             }
         }
+
+        Optional<Page> statisticsPage = extractStatisticsRows(page);
+        statisticsPage.ifPresent(statisticsAggregation::addInput);
+    }
+
+    private static Optional<Page> extractStatisticsRows(Page page)
+    {
+        int statisticsPositionCount = 0;
+        for (int position = 0; position < page.getPositionCount(); position++) {
+            if (isStatisticsPosition(page, position)) {
+                statisticsPositionCount++;
+            }
+        }
+
+        if (statisticsPositionCount == 0) {
+            return Optional.empty();
+        }
+
+        if (statisticsPositionCount == page.getPositionCount()) {
+            return Optional.of(page);
+        }
+
+        int selectedPositionsIndex = 0;
+        int[] selectedPositions = new int[statisticsPositionCount];
+        for (int position = 0; position < page.getPositionCount(); position++) {
+            if (isStatisticsPosition(page, position)) {
+                selectedPositions[selectedPositionsIndex] = position;
+                selectedPositionsIndex++;
+            }
+        }
+
+        Block[] blocks = new Block[page.getChannelCount()];
+        for (int channel = 0; channel < page.getChannelCount(); channel++) {
+            blocks[channel] = page.getBlock(channel).copyPositions(selectedPositions, 0, statisticsPositionCount);
+        }
+        return Optional.of(new Page(statisticsPositionCount, blocks));
+    }
+
+    private static boolean isStatisticsPosition(Page page, int position)
```

Comment:
can you add a comment somewhere explaining the multiplexing that you explained to me in person?